app/src/main/java/com/asfoundation/wallet/iab/presentation/main/MainFragment.kt

```diff
@@ -37,31 +48,65 @@ private fun MainScreen(navController: NavController, purchaseData: PurchaseData?
   val viewModel = rememberMainViewModel(navController)
 
   IAPTheme {
-    IAPBottomSheet(
-      showWalletIcon = false,
-      fullscreen = false,
+    RealMainScreen(
+      showWalletIcon = showWalletIcon,
+    )
+  }
+}
+
+@Composable
+private fun RealMainScreen(
+  modifier: Modifier = Modifier,
+  showWalletIcon: Boolean,
+) {
+  var isExpanded by rememberSaveable { mutableStateOf(false) }
+
+  val onPurchaseInfoClick = { isExpanded = !isExpanded }
+  
+  IAPBottomSheet(
+    modifier = modifier,
+    showWalletIcon = showWalletIcon,
+    fullscreen = false,
+  ) {
+    Column(
+      modifier = Modifier
+        .padding(16.dp)
+        .clip(RoundedCornerShape(12.dp))
+        .background(IAPTheme.colors.primaryContainer)
     ) {
-      GenericError(
-        titleText = "Screen to navigate to Verify",
-        secondaryButtonText = "Navigate to verify",
-        onSecondaryButtonClick = { viewModel.navigateTo(MainFragmentDirections.actionNavigateToVerifyFragment()) },
-        onSupportClick = {},
+      PurchaseInfo(
+        modifier = Modifier.padding(16.dp),
+        purchaseInfo = emptyPurchaseInfo.copy(hasFees = false),
+        isExpanded = false,
+      )
+      SeparatorLine()
+      PurchaseInfo(
+        modifier = Modifier
+          .clickable(onClick = onPurchaseInfoClick)
```

Comment:
Don't forget to implement the test Tag IDs on everything that has a click, so we are ready for testing.